src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonMetadataServicesConverter.cs

```diff
@@ -69,7 +69,6 @@ internal override bool OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializer
                 !state.CurrentContainsMetadata) // Do not use the fast path if state needs to write metadata.
             {
                 Debug.Assert(jsonTypeInfo is JsonTypeInfo<T> typeInfo && typeInfo.SerializeHandler != null);
-                Debug.Assert(jsonTypeInfo.CanUseSerializeHandler);
```

Comment:
Why wouldn't we want to assert this? Related - https://github.com/dotnet/runtime/pull/80741#discussion_r1081087029.

Reply:
It's already checked in the if conditional above, so it's redundant.